### PR TITLE
Do not always return partial_filter_selector

### PR DESCRIPTION
**Pull request by garrensmith**
For Get / _index only return the partial_filter_selector for an index if
it is set to something other than default

## Checklist

- [x] Code is written and works correctly;
- [x] Changes are covered by tests;
- [ ] Documentation reflects the changes;

### CODE REVIEW

File: src/mango/test/16-index-selectors-test.py [hunk 1]
```diff
@@ -161,6 +161,12 @@ def test_saves_partial_filter_selector_in_index(self):
         indexes = self.db.list_indexes()
         self.assertEqual(indexes[1]["def"]["partial_filter_selector"], selector)
 
+    def test_partial_filter_only_in_return_if_not_default(self):
```
Comment (willholley) [inline src/mango/test/16-index-selectors-test.py]: do we have a test for the negative as well (i.e. partial_filter_selector is not present in the json when it has the default value / was not specified)?

File: src/mango/test/16-index-selectors-test.py [hunk 2]
```diff
@@ -264,4 +270,11 @@ def test_text_old_selector_still_supported(self):
         resp = self.db.find(selector, explain=True, use_index='oldschooltext')
         self.assertEqual(resp["index"]["name"], "oldschooltext")
         docs = self.db.find(selector, use_index='oldschooltext')
-        self.assertEqual(len(docs), 3)
\ No newline at end of file
+        self.assertEqual(len(docs), 3)
+
+    @unittest.skipUnless(mango.has_text_service(), "requires text service")
+    def test_text_partial_filter_only_in_return_if_not_default(self):
```
Comment (willholley) [inline src/mango/test/16-index-selectors-test.py]: same as above - do we test the opposite situation?

File: src/mango/src/mango_idx_view.erl [hunk 3]
```diff
@@ -120,7 +120,7 @@ is_usable(Idx, Selector, SortFields) ->
     % and the selector is not a text search (so requires a text index)
     RequiredFields = columns(Idx),
 
-    % sort fields are required to exist in the results so 
+    % sort fields are required to exist in the results so
```
Comment (willholley) [inline src/mango/src/mango_idx_view.erl]: unrelated whitespace change

Comment (garrensmith) [reply src/mango/src/mango_idx_view.erl]: This was my editor removing whitespaces.
